2015-02-04  Nick Clifton  <nickc@redhat.com>

	PR target/64408
	* config/fr30/predicates.md (di_operand): Add SUBREG to the list
	of accepted codes.
	(nonimmediate_di_operand): Likewise.

Index: gcc/config/fr30/predicates.md
===================================================================
--- gcc/config/fr30/predicates.md       (revision 220398)
+++ gcc/config/fr30/predicates.md       (working copy)
@@ -64,7 +64,7 @@
 ;; Returns TRUE if OP is a valid operand of a DImode operation.
 
 (define_predicate "di_operand"
-  (match_code "const_int,const_double,reg,mem")
+  (match_code "const_int,const_double,reg,subreg,mem")
 {
   if (register_operand (op, mode))
     return TRUE;
@@ -92,7 +92,7 @@
 ;; Returns TRUE if OP is a DImode register or MEM.
 
 (define_predicate "nonimmediate_di_operand"
-  (match_code "reg,mem")
+  (match_code "reg,subreg,mem")
 {
   if (register_operand (op, mode))
     return TRUE;
